### PR TITLE
feat: infinity scroll

### PR DESCRIPTION
**Pull request by topether21**
We could load the ordinals by scrolling.

Also we can use a virtual list like:
https://react-window.vercel.app/#/examples/list/variable-size

- Filters will be available when we add them in the backend.
- I added @Uthpala PR https://github.com/dannydeezy/nosft/pull/73

https://user-images.githubusercontent.com/126987350/226481682-0a9a9030-ca76-4bd9-abb0-d251026ced39.mov

### CODE REVIEW

**Comment by Uthpala** [comment]
@topether21 I think you will have to implement the sorting in the frontend. If not when the user triggers a sort we will have to fetch all the inscriptions again for that order. So its best to do the sorting in the frontend.

**Comment by topether21** [comment]
@andredevjs @Uthpala I enable filters and sort.

File: src/hooks/use-ordinals.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
+/* eslint-disable no-restricted-syntax */
+/* eslint-disable no-await-in-loop */
+import { useMemo, useState } from "react";
+import { FETCH_SIZE } from "@lib/constants";
+import { useEffectOnce, useDebounce } from "react-use";
+import { matchSorter } from "match-sorter";
+
+const axios = require("axios");
```
Comment (Uthpala) [inline src/hooks/use-ordinals.js:8]: Why are we importing this the old way, 
```suggestion
import axios from 'axios'
```

**Comment by topether21** [comment]
I made the PR to main. Updating to use develop.

**Comment by Uthpala** [comment]
Okay i just finished reviewing this. All is good to me. But unfortunately i cannot test every functionality to confirm its working 100% 

I tested filtering and sorting and inscription fetching and they are working well. 

Can someone please confirm other functionality like sending an ordinal is working so we can merge this.

**Comment by andredevjs** [comment]
https://github.com/dannydeezy/nosft/pull/73 is merged. please update your branch with main, since you might have conflicts.

**Comment by topether21** [comment]
Very big screens are not triggering on scroll. Debugging...

image

File: src/utils/utxos.js
Comment (Uthpala) [inline src/utils/utxos.js:1]: I think you can delete this file right ?

**Review by Uthpala — APPROVED** [review]

**Review by andredevjs — CHANGES_REQUESTED** [review]
@topether21 on load, it is adding the new inscriptions on top, but it should add them at the end!, just like FB or IG, or Pinterest with their posts. It also be good, if we don't have to scroll all the way down to load more inscriptions, that should happen opportunistically by loading them before the user reaches the end, that improves user experience by not having the user to wait for those inscriptions.

File: package.json [hunk 2]
```diff
@@ -17,6 +17,7 @@
         "copy-webpack-plugin": "^11.0.0",
         "ecpair": "^2.1.0",
         "framer-motion": "^9.0.3",
+        "lodash": "^4.17.21",
```
Comment (andredevjs) [inline package.json:20]: do we need the whole package?

**Comment by topether21** [comment]
image

https://nosft.xyz/api/inscriptions/bc1p6hsehjafcdzzht5shm4qmm3w2aspwlh843w5r9s33ejhhwesy9qss2hxhp?offset=0&limit=5

@andredevjs here we are missing txid, and vout from the new API so we need to add them.
For example it has missing data on Send modal.

image

**Review by andredevjs — APPROVED** [review]

**Pull request merged**